#! /bin/sh -e
## 002_more-i18n.dpatch by  <bubulle@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add new files with translatable strings

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- shadow-4.0.3.orig/po/POTFILES.in
+++ shadow-4.0.3/po/POTFILES.in
@@ -1,5 +1,19 @@
 # List of files which contain translatable strings.
 
+lib/commonio.c
+lib/dialchk.c
+lib/dialup.c
+lib/encrypt.c
+lib/fputsx.c
+lib/getdef.c
+lib/getpass.c
+lib/grdbm.c
+lib/groupio.c
+lib/grpack.c
+lib/gsdbm.c
+lib/gshadow.c
+lib/gspack.c
+lib/lockpw.c
 libmisc/addgrps.c
 libmisc/age.c
 libmisc/basename.c
@@ -13,6 +27,7 @@
 libmisc/env.c
 libmisc/failure.c
 libmisc/fields.c
+libmisc/getdate.c
 libmisc/hushed.c
 libmisc/isexpired.c
 libmisc/limits.c
@@ -25,9 +40,11 @@
 libmisc/mail.c
 libmisc/motd.c
 libmisc/myname.c
+libmisc/nscd.c
 libmisc/obscure.c
 libmisc/pam_pass.c
 libmisc/pwd2spwd.c
+libmisc/pwdcheck.c
 libmisc/pwd_init.c
 libmisc/rlogin.c
 libmisc/salt.c
@@ -44,60 +61,32 @@
 libmisc/utmp.c
 libmisc/valid.c
 libmisc/xmalloc.c
-lib/commonio.c
-lib/dialchk.c
-lib/dialup.c
-lib/encrypt.c
-lib/fputsx.c
-lib/getdef.c
-lib/getpass.c
-lib/grdbm.c
-lib/groupio.c
-lib/grpack.c
-lib/gsdbm.c
-lib/gshadow.c
-lib/gspack.c
-lib/lockpw.c
-lib/md5.c
-lib/md5crypt.c
-lib/mkdir.c
 lib/port.c
-lib/putgrent.c
-lib/putpwent.c
-lib/putspent.c
 lib/pwauth.c
 lib/pwdbm.c
 lib/pwio.c
 lib/pwpack.c
 lib/rad64.c
-lib/rename.c
-lib/rmdir.c
 lib/sgetgrent.c
 lib/sgetpwent.c
-lib/sgetspent.c
 lib/sgroupio.c
 lib/shadow.c
 lib/shadowio.c
-lib/snprintf.c
 lib/spdbm.c
 lib/sppack.c
-lib/strcasecmp.c
-lib/strdup.c
-lib/strerror.c
-lib/strstr.c
 lib/tcfsio.c
 lib/utent.c
 src/chage.c
 src/chfn.c
 src/chpasswd.c
 src/chsh.c
+src/cppw.c
 src/dpasswd.c
 src/expiry.c
 src/faillog.c
 src/gpasswd.c
 src/groupadd.c
 src/groupdel.c
-src/groupmems.c
 src/groupmod.c
 src/groups.c
 src/grpck.c
@@ -114,11 +103,10 @@
 src/pwck.c
 src/pwconv.c
 src/pwunconv.c
-src/su.c
 src/suauth.c
+src/su.c
 src/sulogin.c
 src/useradd.c
 src/userdel.c
 src/usermod.c
 src/vipw.c
-
--- shadow-4.0.3.orig/po/shadow.pot
+++ shadow-4.0.3/po/shadow.pot
@@ -1,18 +1,48 @@
 # SOME DESCRIPTIVE TITLE.
-# Copyright (C) YEAR Free Software Foundation, Inc.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
 # FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
 #
 #, fuzzy
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2002-03-12 12:16+0100\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2004-07-26 06:02-0600\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=CHARSET\n"
 "Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=INTEGER; plural=EXPRESSION;\n"
+
+#: lib/dialchk.c:71
+msgid "Dialup Password: "
+msgstr ""
+
+#: lib/getdef.c:276
+#, c-format
+msgid "Could not allocate space for config info.\n"
+msgstr ""
+
+#.
+#. * Item was never found.
+#.
+#: lib/getdef.c:330
+#, c-format
+msgid "configuration error - unknown item '%s' (notify administrator)\n"
+msgstr ""
+
+#: lib/getdef.c:417
+#, c-format
+msgid "error - lookup '%s' failed\n"
+msgstr ""
+
+#: lib/getdef.c:425
+#, c-format
+msgid "%s not found\n"
+msgstr ""
 
 #: libmisc/addgrps.c:60
 #, c-format
@@ -20,18 +50,22 @@
 msgstr ""
 
 #: libmisc/addgrps.c:71
+#, c-format
 msgid "Warning: too many groups\n"
 msgstr ""
 
 #: libmisc/age.c:77
+#, c-format
 msgid "Your password has expired."
 msgstr ""
 
 #: libmisc/age.c:80
+#, c-format
 msgid "Your password is inactive."
 msgstr ""
 
 #: libmisc/age.c:83
+#, c-format
 msgid "Your login has expired."
 msgstr ""
 
@@ -49,10 +83,12 @@
 msgstr ""
 
 #: libmisc/age.c:178
+#, c-format
 msgid "Your password will expire tomorrow.\n"
 msgstr ""
 
 #: libmisc/age.c:180
+#, c-format
 msgid "Your password will expire today.\n"
 msgstr ""
 
@@ -62,6 +98,7 @@
 msgstr ""
 
 #: libmisc/env.c:160
+#, c-format
 msgid "Environment overflow\n"
 msgstr ""
 
@@ -70,20 +107,15 @@
 msgid "You may not change $%s\n"
 msgstr ""
 
-#: libmisc/failure.c:238
+#: libmisc/failure.c:242
 #, c-format
-msgid "%d %s since last login.  Last was %s on %s.\n"
-msgstr ""
-
-#: libmisc/failure.c:239
-msgid "failures"
-msgstr ""
-
-#: libmisc/failure.c:239
-msgid "failure"
-msgstr ""
+msgid "%d failure since last login.  Last was %s on %s.\n"
+msgid_plural "%d failures since last login.  Last was %s on %s.\n"
+msgstr[0] ""
+msgstr[1] ""
 
 #: libmisc/limits.c:397
+#, c-format
 msgid "Too many logins.\n"
 msgstr ""
 
@@ -124,6 +156,11 @@
 msgid "passwd: %s\n"
 msgstr ""
 
+#: libmisc/pwdcheck.c:34 libmisc/pwdcheck.c:64
+#, c-format
+msgid "Incorrect password for %s.\n"
+msgstr ""
+
 #: libmisc/setupenv.c:211
 #, c-format
 msgid "Unable to cd to \"%s\"\n"
@@ -158,37 +195,11 @@
 msgid "Can't change root directory to \"%s\"\n"
 msgstr ""
 
-#: libmisc/xmalloc.c:28
+#: libmisc/xmalloc.c:26
 #, c-format
 msgid "malloc(%d) failed\n"
 msgstr ""
 
-#: lib/dialchk.c:71
-msgid "Dialup Password: "
-msgstr ""
-
-#: lib/getdef.c:276
-msgid "Could not allocate space for config info.\n"
-msgstr ""
-
-#.
-#. * Item was never found.
-#.
-#: lib/getdef.c:330
-#, c-format
-msgid "configuration error - unknown item '%s' (notify administrator)\n"
-msgstr ""
-
-#: lib/getdef.c:417
-#, c-format
-msgid "error - lookup '%s' failed\n"
-msgstr ""
-
-#: lib/getdef.c:425
-#, c-format
-msgid "%s not found\n"
-msgstr ""
-
 #.
 #. * get the password from her, and set the salt for
 #. * the decryption from the group file.
@@ -206,50 +217,46 @@
 msgid "(Echo on) "
 msgstr ""
 
-#: lib/strerror.c:20
-#, c-format
-msgid "Unknown error %d"
-msgstr ""
-
-#: src/chage.c:116
+#: src/chage.c:118
 #, c-format
 msgid ""
 "Usage: %s [-l] [-m min_days] [-M max_days] [-W warn]\n"
 "\t[-I inactive] [-E expire] [-d last_day] user\n"
 msgstr ""
 
-#: src/chage.c:122
+#: src/chage.c:124
 #, c-format
 msgid "Usage: %s [-l] [-m min_days] [-M max_days] [-d last_day] user\n"
 msgstr ""
 
-#: src/chage.c:157
+#: src/chage.c:159
+#, c-format
 msgid ""
 "Enter the new value, or press ENTER for the default\n"
 "\n"
 msgstr ""
 
-#: src/chage.c:160
+#: src/chage.c:162
 msgid "Minimum Password Age"
 msgstr ""
 
-#: src/chage.c:166
+#: src/chage.c:168
 msgid "Maximum Password Age"
 msgstr ""
 
-#: src/chage.c:174
+#: src/chage.c:176
 msgid "Last Password Change (YYYY-MM-DD)"
 msgstr ""
 
-#: src/chage.c:183
+#: src/chage.c:185
 msgid "Password Expiration Warning"
 msgstr ""
 
-#: src/chage.c:189
+#: src/chage.c:191
 msgid "Password Inactive"
 msgstr ""
 
-#: src/chage.c:197
+#: src/chage.c:199
 msgid "Account Expiration Date (YYYY-MM-DD)"
 msgstr ""
 
@@ -260,22 +267,22 @@
 #. * expires that the user is told, and the number of days after the
 #. * password expires that the account becomes unusable.
 #.
-#: src/chage.c:248
+#: src/chage.c:250
 #, c-format
 msgid "Minimum:\t%ld\n"
 msgstr ""
 
-#: src/chage.c:249
+#: src/chage.c:251
 #, c-format
 msgid "Maximum:\t%ld\n"
 msgstr ""
 
-#: src/chage.c:251
+#: src/chage.c:253
 #, c-format
 msgid "Warning:\t%ld\n"
 msgstr ""
 
-#: src/chage.c:252
+#: src/chage.c:254
 #, c-format
 msgid "Inactive:\t%ld\n"
 msgstr ""
@@ -284,11 +291,13 @@
 #. * The "last change" date is either "Never" or the date the password
 #. * was last modified. The date is the number of days since 1/1/1970.
 #.
-#: src/chage.c:260
+#: src/chage.c:262
+#, c-format
 msgid "Last Change:\t\t"
 msgstr ""
 
-#: src/chage.c:262 src/chage.c:276 src/chage.c:293 src/chage.c:306
+#: src/chage.c:264 src/chage.c:278 src/chage.c:295 src/chage.c:308
+#, c-format
 msgid "Never\n"
 msgstr ""
 
@@ -296,7 +305,8 @@
 #. * The password expiration date is determined from the last change
 #. * date plus the number of days the password is valid for.
 #.
-#: src/chage.c:273
+#: src/chage.c:275
+#, c-format
 msgid "Password Expires:\t"
 msgstr ""
 
@@ -306,7 +316,8 @@
 #. * number of inactive days is added. The resulting date is when the
 #. * active will be disabled.
 #.
-#: src/chage.c:290
+#: src/chage.c:292
+#, c-format
 msgid "Password Inactive:\t"
 msgstr ""
 
@@ -314,90 +325,92 @@
 #. * The account will expire on the given date regardless of the
 #. * password expiring or not.
 #.
-#: src/chage.c:304
+#: src/chage.c:306
+#, c-format
 msgid "Account Expires:\t"
 msgstr ""
 
-#: src/chage.c:468
+#: src/chage.c:470
 #, c-format
 msgid "%s: do not include \"l\" with other flags\n"
 msgstr ""
 
-#: src/chage.c:481 src/chage.c:652 src/login.c:518
+#: src/chage.c:483 src/chage.c:654 src/login.c:530
 #, c-format
 msgid "%s: permission denied\n"
 msgstr ""
 
-#: src/chage.c:512 src/chpasswd.c:139 src/groupadd.c:529 src/groupdel.c:322
-#: src/groupmod.c:520 src/newusers.c:381 src/useradd.c:1754 src/userdel.c:717
-#: src/usermod.c:1732
+#: src/chage.c:514 src/chpasswd.c:141 src/groupadd.c:531 src/groupdel.c:324
+#: src/groupmod.c:522 src/newusers.c:383 src/useradd.c:1756 src/userdel.c:728
+#: src/usermod.c:1739
 #, c-format
 msgid "%s: PAM authentication failed\n"
 msgstr ""
 
-#: src/chage.c:533 src/chpasswd.c:157
+#: src/chage.c:535 src/chpasswd.c:159
 #, c-format
 msgid "%s: can't lock password file\n"
 msgstr ""
 
-#: src/chage.c:544 src/chpasswd.c:162
+#: src/chage.c:546 src/chpasswd.c:164
 #, c-format
 msgid "%s: can't open password file\n"
 msgstr ""
 
-#: src/chage.c:552
+#: src/chage.c:554
 #, c-format
 msgid "%s: unknown user: %s\n"
 msgstr ""
 
-#: src/chage.c:571
+#: src/chage.c:573
 #, c-format
 msgid "%s: can't lock shadow password file"
 msgstr ""
 
-#: src/chage.c:579
+#: src/chage.c:581
 #, c-format
 msgid "%s: can't open shadow password file"
 msgstr ""
 
-#: src/chage.c:674
+#: src/chage.c:676
 #, c-format
 msgid "Changing the aging information for %s\n"
 msgstr ""
 
-#: src/chage.c:677
+#: src/chage.c:679
 #, c-format
 msgid "%s: error changing fields\n"
 msgstr ""
 
-#: src/chage.c:705 src/chage.c:771 src/pwunconv.c:174
+#: src/chage.c:707 src/chage.c:773 src/pwunconv.c:174
 #, c-format
 msgid "%s: can't update password file\n"
 msgstr ""
 
-#: src/chage.c:737 src/pwunconv.c:168
+#: src/chage.c:739 src/pwunconv.c:168
 #, c-format
 msgid "%s: can't update shadow password file\n"
 msgstr ""
 
-#: src/chage.c:789 src/chage.c:805 src/chfn.c:570 src/chsh.c:382
+#: src/chage.c:791 src/chage.c:807 src/chfn.c:570 src/chsh.c:382
 #: src/passwd.c:747 src/passwd.c:859
+#, c-format
 msgid "Error updating the DBM password entry.\n"
 msgstr ""
 
-#: src/chage.c:823
+#: src/chage.c:825
 #, c-format
 msgid "%s: can't rewrite shadow password file\n"
 msgstr ""
 
-#: src/chage.c:838
+#: src/chage.c:840
 #, c-format
 msgid "%s: can't rewrite password file\n"
 msgstr ""
 
-#: src/chage.c:857 src/chpasswd.c:343 src/groupadd.c:590 src/groupdel.c:409
-#: src/groupmod.c:604 src/newusers.c:628 src/useradd.c:1869 src/userdel.c:849
-#: src/usermod.c:1811
+#: src/chage.c:859 src/chpasswd.c:345 src/groupadd.c:592 src/groupdel.c:411
+#: src/groupmod.c:606 src/newusers.c:630 src/useradd.c:1871 src/userdel.c:860
+#: src/usermod.c:1818
 #, c-format
 msgid "%s: PAM chauthtok failed\n"
 msgstr ""
@@ -415,6 +428,7 @@
 msgstr ""
 
 #: src/chfn.c:139
+#, c-format
 msgid "Enter the new value, or press ENTER for the default\n"
 msgstr ""
 
@@ -520,81 +534,87 @@
 msgstr ""
 
 #: src/chfn.c:504 src/chsh.c:316 src/gpasswd.c:570 src/passwd.c:1277
+#, c-format
 msgid "Cannot change ID to root.\n"
 msgstr ""
 
 #: src/chfn.c:519 src/chsh.c:331 src/passwd.c:656 src/passwd.c:811
+#, c-format
 msgid "Cannot lock the password file; try again later.\n"
 msgstr ""
 
 #: src/chfn.c:525 src/chsh.c:337 src/passwd.c:661 src/passwd.c:816
+#, c-format
 msgid "Cannot open the password file.\n"
 msgstr ""
 
-#: src/chfn.c:542 src/chsh.c:352 src/passwd.c:667 src/usermod.c:1348
+#: src/chfn.c:542 src/chsh.c:352 src/passwd.c:667 src/usermod.c:1350
 #, c-format
 msgid "%s: %s not found in /etc/passwd\n"
 msgstr ""
 
 #: src/chfn.c:561 src/chsh.c:373 src/passwd.c:740 src/passwd.c:852
+#, c-format
 msgid "Error updating the password entry.\n"
 msgstr ""
 
 #: src/chfn.c:585 src/chsh.c:397 src/passwd.c:755 src/passwd.c:867
+#, c-format
 msgid "Cannot commit password file changes.\n"
 msgstr ""
 
 #: src/chfn.c:592 src/chsh.c:404
+#, c-format
 msgid "Cannot unlock the password file.\n"
 msgstr ""
 
-#: src/chpasswd.c:66
+#: src/chpasswd.c:68
 #, c-format
 msgid "usage: %s [-e]\n"
 msgstr ""
 
-#: src/chpasswd.c:171 src/pwconv.c:94
+#: src/chpasswd.c:173 src/pwconv.c:94
 #, c-format
 msgid "%s: can't lock shadow file\n"
 msgstr ""
 
-#: src/chpasswd.c:177 src/gpasswd.c:596 src/pwconv.c:99 src/pwunconv.c:103
+#: src/chpasswd.c:179 src/gpasswd.c:596 src/pwconv.c:99 src/pwunconv.c:103
 #: src/pwunconv.c:108
 #, c-format
 msgid "%s: can't open shadow file\n"
 msgstr ""
 
-#: src/chpasswd.c:200 src/newusers.c:460
+#: src/chpasswd.c:202 src/newusers.c:462
 #, c-format
 msgid "%s: line %d: line too long\n"
 msgstr ""
 
-#: src/chpasswd.c:220
+#: src/chpasswd.c:222
 #, c-format
 msgid "%s: line %d: missing new password\n"
 msgstr ""
 
-#: src/chpasswd.c:237
+#: src/chpasswd.c:239
 #, c-format
 msgid "%s: line %d: unknown user %s\n"
 msgstr ""
 
-#: src/chpasswd.c:290
+#: src/chpasswd.c:292
 #, c-format
 msgid "%s: line %d: cannot update password entry\n"
 msgstr ""
 
-#: src/chpasswd.c:307 src/newusers.c:588
+#: src/chpasswd.c:309 src/newusers.c:590
 #, c-format
 msgid "%s: error detected, changes ignored\n"
 msgstr ""
 
-#: src/chpasswd.c:319
+#: src/chpasswd.c:321
 #, c-format
 msgid "%s: error updating shadow file\n"
 msgstr ""
 
-#: src/chpasswd.c:328
+#: src/chpasswd.c:330
 #, c-format
 msgid "%s: error updating password file\n"
 msgstr ""
@@ -605,6 +625,7 @@
 msgstr ""
 
 #: src/chsh.c:89
+#, c-format
 msgid "Enter the new value, or press return for the default\n"
 msgstr ""
 
@@ -666,6 +687,7 @@
 msgstr ""
 
 #: src/expiry.c:59
+#, c-format
 msgid "Usage: expiry {-f|-c}\n"
 msgstr ""
 
@@ -690,6 +712,7 @@
 msgstr ""
 
 #: src/faillog.c:212
+#, c-format
 msgid "Username   Failures  Maximum  Latest\n"
 msgstr ""
 
@@ -739,6 +762,7 @@
 msgstr ""
 
 #: src/gpasswd.c:157
+#, c-format
 msgid "Permission denied.\n"
 msgstr ""
 
@@ -748,6 +772,7 @@
 msgstr ""
 
 #: src/gpasswd.c:294
+#, c-format
 msgid "Who are you?\n"
 msgstr ""
 
@@ -854,33 +879,34 @@
 msgid "%s: can't update DBM shadow files\n"
 msgstr ""
 
-#: src/groupadd.c:101
+#: src/groupadd.c:103
+#, c-format
 msgid "usage: groupadd [-g gid [-o]] group\n"
 msgstr ""
 
-#: src/groupadd.c:167 src/groupadd.c:192 src/groupmod.c:179 src/groupmod.c:228
-#: src/useradd.c:970 src/usermod.c:552 src/usermod.c:695
+#: src/groupadd.c:169 src/groupadd.c:194 src/groupmod.c:181 src/groupmod.c:230
+#: src/useradd.c:972 src/usermod.c:554 src/usermod.c:697
 #, c-format
 msgid "%s: error adding new group entry\n"
 msgstr ""
 
-#: src/groupadd.c:178 src/groupadd.c:203 src/groupmod.c:197 src/useradd.c:981
-#: src/usermod.c:564 src/usermod.c:708
+#: src/groupadd.c:180 src/groupadd.c:205 src/groupmod.c:199 src/useradd.c:983
+#: src/usermod.c:566 src/usermod.c:710
 #, c-format
 msgid "%s: cannot add new dbm group entry\n"
 msgstr ""
 
-#: src/groupadd.c:255 src/useradd.c:1034
+#: src/groupadd.c:257 src/useradd.c:1036
 #, c-format
 msgid "%s: name %s is not unique\n"
 msgstr ""
 
-#: src/groupadd.c:270
+#: src/groupadd.c:272
 #, c-format
 msgid "%s: gid %u is not unique\n"
 msgstr ""
 
-#: src/groupadd.c:294
+#: src/groupadd.c:296
 #, c-format
 msgid "%s: can't get unique gid\n"
 msgstr ""
@@ -888,78 +914,79 @@
 #.
 #. * All invalid group names land here.
 #.
-#: src/groupadd.c:317 src/groupmod.c:341
+#: src/groupadd.c:319 src/groupmod.c:343
 #, c-format
 msgid "%s: %s is a not a valid group name\n"
 msgstr ""
 
-#: src/groupadd.c:346 src/groupmod.c:366
+#: src/groupadd.c:348 src/groupmod.c:368
 #, c-format
 msgid "%s: invalid group %s\n"
 msgstr ""
 
-#: src/groupadd.c:363 src/useradd.c:1313
+#: src/groupadd.c:365 src/useradd.c:1315
 #, c-format
 msgid "%s: -O requires NAME=VALUE\n"
 msgstr ""
 
-#: src/groupadd.c:407 src/groupdel.c:173 src/groupmod.c:401 src/useradd.c:1428
-#: src/userdel.c:312 src/usermod.c:576
+#: src/groupadd.c:409 src/groupdel.c:175 src/groupmod.c:403 src/useradd.c:1430
+#: src/userdel.c:323 src/usermod.c:578
 #, c-format
 msgid "%s: cannot rewrite group file\n"
 msgstr ""
 
-#: src/groupadd.c:415 src/groupdel.c:179 src/groupmod.c:409 src/useradd.c:1437
-#: src/userdel.c:319 src/usermod.c:722
+#: src/groupadd.c:417 src/groupdel.c:181 src/groupmod.c:411 src/useradd.c:1439
+#: src/userdel.c:330 src/usermod.c:724
 #, c-format
 msgid "%s: cannot rewrite shadow group file\n"
 msgstr ""
 
-#: src/groupadd.c:433 src/groupdel.c:198 src/groupmod.c:427 src/userdel.c:403
+#: src/groupadd.c:435 src/groupdel.c:200 src/groupmod.c:429 src/userdel.c:414
 #, c-format
 msgid "%s: unable to lock group file\n"
 msgstr ""
 
-#: src/groupadd.c:438 src/groupdel.c:202 src/groupmod.c:432
+#: src/groupadd.c:440 src/groupdel.c:204 src/groupmod.c:434
 #, c-format
 msgid "%s: unable to open group file\n"
 msgstr ""
 
-#: src/groupadd.c:445 src/groupdel.c:207 src/groupmod.c:439 src/userdel.c:414
+#: src/groupadd.c:447 src/groupdel.c:209 src/groupmod.c:441 src/userdel.c:425
 #, c-format
 msgid "%s: unable to lock shadow group file\n"
 msgstr ""
 
-#: src/groupadd.c:451 src/groupdel.c:212 src/groupmod.c:445
+#: src/groupadd.c:453 src/groupdel.c:214 src/groupmod.c:447
 #, c-format
 msgid "%s: unable to open shadow group file\n"
 msgstr ""
 
-#: src/groupadd.c:562
+#: src/groupadd.c:564
 #, c-format
 msgid "%s: group %s exists\n"
 msgstr ""
 
-#: src/groupdel.c:92
+#: src/groupdel.c:94
+#, c-format
 msgid "usage: groupdel group\n"
 msgstr ""
 
-#: src/groupdel.c:110 src/groupmod.c:184 src/groupmod.c:233
+#: src/groupdel.c:112 src/groupmod.c:186 src/groupmod.c:235
 #, c-format
 msgid "%s: error removing group entry\n"
 msgstr ""
 
-#: src/groupdel.c:122 src/groupmod.c:204
+#: src/groupdel.c:124 src/groupmod.c:206
 #, c-format
 msgid "%s: error removing group dbm entry\n"
 msgstr ""
 
-#: src/groupdel.c:137
+#: src/groupdel.c:139
 #, c-format
 msgid "%s: error removing shadow group entry\n"
 msgstr ""
 
-#: src/groupdel.c:150 src/groupmod.c:254
+#: src/groupdel.c:152 src/groupmod.c:256
 #, c-format
 msgid "%s: error removing shadow group dbm entry\n"
 msgstr ""
@@ -967,82 +994,47 @@
 #.
 #. * Can't remove the group.
 #.
-#: src/groupdel.c:254
+#: src/groupdel.c:256
 #, c-format
 msgid "%s: cannot remove user's primary group.\n"
 msgstr ""
 
-#: src/groupdel.c:355 src/groupmod.c:550
+#: src/groupdel.c:357 src/groupmod.c:552
 #, c-format
 msgid "%s: group %s does not exist\n"
 msgstr ""
 
-#: src/groupdel.c:369 src/groupmod.c:566
+#: src/groupdel.c:371 src/groupmod.c:568
 #, c-format
 msgid "%s: group %s is a NIS group\n"
 msgstr ""
 
-#: src/groupdel.c:375 src/groupmod.c:571 src/userdel.c:776 src/usermod.c:1045
+#: src/groupdel.c:377 src/groupmod.c:573 src/userdel.c:787 src/usermod.c:1047
 #, c-format
 msgid "%s: %s is the NIS master\n"
 msgstr ""
 
-#: src/groupmems.c:95
-msgid "Member already exists\n"
-msgstr ""
-
-#: src/groupmems.c:125
-msgid "Member to remove could not be found\n"
-msgstr ""
-
-#: src/groupmems.c:158
-msgid "usage: groupmems -a username | -d username | -D | -l [-g groupname]\n"
-msgstr ""
-
-#: src/groupmems.c:201
-msgid "Only root can add members to different groups\n"
-msgstr ""
-
-#: src/groupmems.c:206
-msgid "Group access is required\n"
-msgstr ""
-
-#: src/groupmems.c:210
-msgid "Not primary owner of current group\n"
-msgstr ""
-
-#: src/groupmems.c:215
-msgid "Unable to lock group file\n"
-msgstr ""
-
-#: src/groupmems.c:220
-msgid "Unable to open group file\n"
-msgstr ""
-
-#: src/groupmems.c:240
-msgid "Cannot close group file\n"
-msgstr ""
-
-#: src/groupmod.c:103
+#: src/groupmod.c:105
+#, c-format
 msgid "usage: groupmod [-g gid [-o]] [-n name] group\n"
 msgstr ""
 
-#: src/groupmod.c:161
+#: src/groupmod.c:163
 #, c-format
 msgid "%s: %s not found in /etc/group\n"
 msgstr ""
 
-#: src/groupmod.c:247
+#: src/groupmod.c:249
 #, c-format
 msgid "%s: cannot add new dbm shadow group entry\n"
 msgstr ""
 
-#: src/groupmod.c:300
+#: src/groupmod.c:302
 #, c-format
 msgid "%s: %u is not a unique gid\n"
 msgstr ""
 
-#: src/groupmod.c:330
+#: src/groupmod.c:332
 #, c-format
 msgid "%s: %s is not a unique name\n"
 msgstr ""
@@ -1066,17 +1058,17 @@
 msgid "No"
 msgstr ""
 
-#: src/grpck.c:192 src/pwck.c:175
+#: src/grpck.c:196 src/pwck.c:175
 #, c-format
 msgid "%s: -s and -r are incompatibile\n"
 msgstr ""
 
-#: src/grpck.c:232 src/grpck.c:242 src/pwck.c:215 src/pwck.c:225
+#: src/grpck.c:236 src/grpck.c:246 src/pwck.c:215 src/pwck.c:225
 #, c-format
 msgid "%s: cannot lock file %s\n"
 msgstr ""
 
-#: src/grpck.c:259 src/grpck.c:268 src/mkpasswd.c:202 src/pwck.c:242
+#: src/grpck.c:263 src/grpck.c:272 src/mkpasswd.c:202 src/pwck.c:242
 #: src/pwck.c:251
 #, c-format
 msgid "%s: cannot open file %s\n"
@@ -1086,12 +1078,13 @@
 #. * Tell the user this entire line is bogus and ask
 #. * them to delete it.
 #.
-#: src/grpck.c:311
+#: src/grpck.c:315
+#, c-format
 msgid "invalid group file entry\n"
 msgstr ""
 
-#: src/grpck.c:312 src/grpck.c:375 src/grpck.c:458 src/grpck.c:521
-#: src/grpck.c:538 src/pwck.c:296 src/pwck.c:358 src/pwck.c:457 src/pwck.c:519
+#: src/grpck.c:316 src/grpck.c:383 src/grpck.c:466 src/grpck.c:529
+#: src/grpck.c:546 src/pwck.c:296 src/pwck.c:358 src/pwck.c:457 src/pwck.c:519
 #: src/pwck.c:543
 #, c-format
 msgid "delete line `%s'? "
@@ -1101,21 +1094,21 @@
 #. * Tell the user this entry is a duplicate of
 #. * another and ask them to delete it.
 #.
-#: src/grpck.c:374
+#: src/grpck.c:382
 msgid "duplicate group entry\n"
 msgstr ""
 
-#: src/grpck.c:391
+#: src/grpck.c:399
 #, c-format
 msgid "invalid group name `%s'\n"
 msgstr ""
 
-#: src/grpck.c:418
+#: src/grpck.c:426
 #, c-format
 msgid "group %s: no user %s\n"
 msgstr ""
 
-#: src/grpck.c:420 src/grpck.c:590
+#: src/grpck.c:428 src/grpck.c:598
 #, c-format
 msgid "delete member `%s'? "
 msgstr ""
@@ -1124,7 +1117,8 @@
 #. * Tell the user this entire line is bogus and ask
 #. * them to delete it.
 #.
-#: src/grpck.c:457
+#: src/grpck.c:465
+#, c-format
 msgid "invalid shadow group file entry\n"
 msgstr ""
 
@@ -1132,45 +1126,45 @@
 #. * Tell the user this entry is a duplicate of
 #. * another and ask them to delete it.
 #.
-#: src/grpck.c:520
+#: src/grpck.c:528
 msgid "duplicate shadow group entry\n"
 msgstr ""
 
-#: src/grpck.c:537
+#: src/grpck.c:545
 msgid "no matching group file entry\n"
 msgstr ""
 
-#: src/grpck.c:558
+#: src/grpck.c:566
 #, c-format
 msgid "shadow group %s: no administrative user %s\n"
 msgstr ""
 
-#: src/grpck.c:560
+#: src/grpck.c:568
 #, c-format
 msgid "delete administrative member `%s'? "
 msgstr ""
 
-#: src/grpck.c:588
+#: src/grpck.c:596
 #, c-format
 msgid "shadow group %s: no user %s\n"
 msgstr ""
 
-#: src/grpck.c:616 src/grpck.c:622 src/pwck.c:577 src/pwck.c:585
+#: src/grpck.c:624 src/grpck.c:630 src/pwck.c:577 src/pwck.c:585
 #, c-format
 msgid "%s: cannot update file %s\n"
 msgstr ""
 
-#: src/grpck.c:647 src/pwck.c:612
+#: src/grpck.c:655 src/pwck.c:612
 #, c-format
 msgid "%s: the files have been updated; run mkpasswd\n"
 msgstr ""
 
-#: src/grpck.c:648 src/grpck.c:652 src/pwck.c:613 src/pwck.c:617
+#: src/grpck.c:656 src/grpck.c:660 src/pwck.c:613 src/pwck.c:617
 #, c-format
 msgid "%s: no changes\n"
 msgstr ""
 
-#: src/grpck.c:651 src/pwck.c:616
+#: src/grpck.c:659 src/pwck.c:616
 #, c-format
 msgid "%s: the files have been updated\n"
 msgstr ""
@@ -1231,10 +1225,12 @@
 msgstr ""
 
 #: src/id.c:53
+#, c-format
 msgid "usage: id [-a]\n"
 msgstr ""
 
 #: src/id.c:55
+#, c-format
 msgid "usage: id\n"
 msgstr ""
 
@@ -1247,6 +1243,7 @@
 #. * corresponding name for each respective numerical value.
 #.
 #: src/id.c:169
+#, c-format
 msgid " groups="
 msgstr ""
 
@@ -1256,100 +1253,131 @@
 msgstr ""
 
 #: src/lastlog.c:181
+#, c-format
 msgid "Username         Port     From             Latest\n"
 msgstr ""
 
 #: src/lastlog.c:183
+#, c-format
 msgid "Username                Port     Latest\n"
 msgstr ""
 
-#: src/lastlog.c:197
+#: src/lastlog.c:203
 msgid "**Never logged in**"
 msgstr ""
 
-#: src/login.c:185
+#: src/login.c:190
 #, c-format
 msgid "usage: %s [-p] [name]\n"
 msgstr ""
 
-#: src/login.c:188
+#: src/login.c:193
 #, c-format
 msgid "       %s [-p] [-h host] [-f name]\n"
 msgstr ""
 
-#: src/login.c:190
+#: src/login.c:195
 #, c-format
 msgid "       %s [-p] -r host\n"
 msgstr ""
 
-#: src/login.c:274
+#: src/login.c:279
+#, c-format
 msgid "Invalid login time\n"
 msgstr ""
 
-#: src/login.c:329
+#: src/login.c:336
+#, c-format
 msgid ""
 "\n"
 "System closed for routine maintenance\n"
 msgstr ""
 
-#: src/login.c:340
+#: src/login.c:347
+#, c-format
 msgid ""
 "\n"
 "[Disconnect bypassed -- root login allowed.]\n"
 msgstr ""
 
-#: src/login.c:377
+#: src/login.c:384
 #, c-format
 msgid ""
 "\n"
 "Login timed out after %d seconds.\n"
 msgstr ""
 
-#: src/login.c:689
+#: src/login.c:701
 #, c-format
 msgid " on `%.100s' from `%.200s'"
 msgstr ""
 
-#: src/login.c:693
+#: src/login.c:705
 #, c-format
 msgid " on `%.100s'"
 msgstr ""
 
-#: src/login.c:852
+#: src/login.c:810
+#, c-format
+msgid "TOO MANY LOGIN TRIES (%d)%s FOR `%s'"
+msgstr ""
+
+#: src/login.c:817
+#, c-format
+msgid "Maximum number of tries exceeded (%d)\n"
+msgstr ""
+
+#. Serious problems, quit now
+#: src/login.c:823
+#, c-format
+msgid "login: abort requested by PAM\n"
+msgstr ""
+
+#: src/login.c:824
+msgid "PAM_ABORT returned from pam_authenticate()"
+msgstr ""
+
+#: src/login.c:828
+#, c-format
+msgid "FAILED LOGIN (%d)%s FOR `%s', %s"
+msgstr ""
+
+#: src/login.c:924
 #, c-format
 msgid ""
 "\n"
 "%s login: "
 msgstr ""
 
-#: src/login.c:855
+#: src/login.c:927
 msgid "login: "
 msgstr ""
 
-#: src/login.c:1071 src/sulogin.c:233
+#: src/login.c:1154 src/sulogin.c:233
 msgid "Login incorrect"
 msgstr ""
 
-#: src/login.c:1265
+#: src/login.c:1382
 msgid "Warning: login re-enabled after temporary lockout.\n"
 msgstr ""
 
-#: src/login.c:1278
+#: src/login.c:1396
 #, c-format
 msgid "Last login: %s on %s"
 msgstr ""
 
-#: src/login.c:1281
+#: src/login.c:1399
 #, c-format
 msgid "Last login: %.19s on %s"
 msgstr ""
 
-#: src/login.c:1287
+#: src/login.c:1405
 #, c-format
 msgid " from %.*s"
 msgstr ""
 
-#: src/login.c:1364
+#: src/login.c:1451
+#, c-format
 msgid "Starting rad_login\n"
 msgstr ""
 
@@ -1374,6 +1402,7 @@
 msgstr ""
 
 #: src/mkpasswd.c:288
+#, c-format
 msgid " is too long\n"
 msgstr ""
 
@@ -1383,11 +1412,13 @@
 msgstr ""
 
 #: src/mkpasswd.c:323 src/mkpasswd.c:328 src/mkpasswd.c:333 src/mkpasswd.c:338
+#, c-format
 msgid "adding record for name "
 msgstr ""
 
 #: src/mkpasswd.c:324 src/mkpasswd.c:329 src/mkpasswd.c:334 src/mkpasswd.c:339
 #: src/mkpasswd.c:345 src/mkpasswd.c:351 src/mkpasswd.c:356 src/mkpasswd.c:361
+#, c-format
 msgid "\n"
 msgstr ""
 
@@ -1417,10 +1448,12 @@
 msgstr ""
 
 #: src/newgrp.c:61
+#, c-format
 msgid "usage: newgrp [-] [group]\n"
 msgstr ""
 
 #: src/newgrp.c:63
+#, c-format
 msgid "usage: sg group [[-c] command]\n"
 msgstr ""
 
@@ -1438,71 +1471,72 @@
 msgid "Sorry.\n"
 msgstr ""
 
-#: src/newgrp.c:472
+#: src/newgrp.c:477
+#, c-format
 msgid "too many groups\n"
 msgstr ""
 
-#: src/newusers.c:78
+#: src/newusers.c:80
 #, c-format
 msgid "Usage: %s [input]\n"
 msgstr ""
 
-#: src/newusers.c:407
+#: src/newusers.c:409
 #, c-format
 msgid "%s: can't lock /etc/passwd.\n"
 msgstr ""
 
-#: src/newusers.c:419
+#: src/newusers.c:421
 #, c-format
 msgid "%s: can't lock files, try again later\n"
 msgstr ""
 
-#: src/newusers.c:435
+#: src/newusers.c:437
 #, c-format
 msgid "%s: can't open files\n"
 msgstr ""
 
-#: src/newusers.c:480
+#: src/newusers.c:482
 #, c-format
 msgid "%s: line %d: invalid line\n"
 msgstr ""
 
-#: src/newusers.c:499
+#: src/newusers.c:501
 #, c-format
 msgid "%s: line %d: can't create GID\n"
 msgstr ""
 
-#: src/newusers.c:515
+#: src/newusers.c:517
 #, c-format
 msgid "%s: line %d: can't create UID\n"
 msgstr ""
 
-#: src/newusers.c:528
+#: src/newusers.c:530
 #, c-format
 msgid "%s: line %d: cannot find user %s\n"
 msgstr ""
 
-#: src/newusers.c:537
+#: src/newusers.c:539
 #, c-format
 msgid "%s: line %d: can't update password\n"
 msgstr ""
 
-#: src/newusers.c:555
+#: src/newusers.c:557
 #, c-format
 msgid "%s: line %d: mkdir failed\n"
 msgstr ""
 
-#: src/newusers.c:561
+#: src/newusers.c:563
 #, c-format
 msgid "%s: line %d: chown failed\n"
 msgstr ""
 
-#: src/newusers.c:571
+#: src/newusers.c:573
 #, c-format
 msgid "%s: line %d: can't update entry\n"
 msgstr ""
 
-#: src/newusers.c:603
+#: src/newusers.c:605
 #, c-format
 msgid "%s: error updating files\n"
 msgstr ""
@@ -1528,6 +1562,7 @@
 msgstr ""
 
 #: src/passwd.c:286
+#, c-format
 msgid "You can use -t option to force the change.\n"
 msgstr ""
 
@@ -1557,16 +1592,19 @@
 msgstr ""
 
 #: src/passwd.c:350
+#, c-format
 msgid "Try again.\n"
 msgstr ""
 
 #: src/passwd.c:361
+#, c-format
 msgid ""
 "\n"
 "Warning: weak password (enter it again to use it anyway).\n"
 msgstr ""
 
 #: src/passwd.c:371
+#, c-format
 msgid "They don't match; try again.\n"
 msgstr ""
 
@@ -1586,18 +1624,22 @@
 msgstr ""
 
 #: src/passwd.c:769
+#, c-format
 msgid "Cannot lock the TCFS key database; try again later\n"
 msgstr ""
 
 #: src/passwd.c:777
+#, c-format
 msgid "Cannot open the TCFS key database.\n"
 msgstr ""
 
 #: src/passwd.c:785
+#, c-format
 msgid "Error updating the TCFS key database.\n"
 msgstr ""
 
 #: src/passwd.c:792
+#, c-format
 msgid "Cannot commit TCFS changes.\n"
 msgstr ""
 
@@ -1618,7 +1660,7 @@
 
 #: src/passwd.c:1176
 #, c-format
-msgid "You may not change the password for %s.\n"
+msgid "%s: You may not view or modify password information for %s.\n"
 msgstr ""
 
 #: src/passwd.c:1242
@@ -1632,6 +1674,7 @@
 msgstr ""
 
 #: src/passwd.c:1297
+#, c-format
 msgid "Password changed.\n"
 msgstr ""
 
@@ -1650,6 +1693,7 @@
 #. * them to delete it.
 #.
 #: src/pwck.c:295
+#, c-format
 msgid "invalid password file entry\n"
 msgstr ""
 
@@ -1692,6 +1736,7 @@
 #. * them to delete it.
 #.
 #: src/pwck.c:456
+#, c-format
 msgid "invalid shadow password file entry\n"
 msgstr ""
 
@@ -1761,60 +1806,85 @@
 msgid "%s: can't delete shadow password file\n"
 msgstr ""
 
-#: src/su.c:133
+#: src/suauth.c:100
+#, c-format
+msgid "Access to su to that account DENIED.\n"
+msgstr ""
+
+#: src/suauth.c:107
+#, c-format
+msgid "Password authentication bypassed.\n"
+msgstr ""
+
+#: src/suauth.c:114
+#, c-format
+msgid "Please enter your OWN password as authentication.\n"
+msgstr ""
+
+#: src/su.c:217 src/su.c:226
+#, c-format
+msgid "Command line args too long\n"
+msgstr ""
+
+#: src/su.c:239
+#, c-format
+msgid "No shell\n"
+msgstr ""
+
+#: src/su.c:254
 msgid "Sorry."
 msgstr ""
 
-#: src/su.c:216
+#: src/su.c:340
 #, c-format
 msgid "%s: must be run from a terminal\n"
 msgstr ""
 
-#: src/su.c:306
+#: src/su.c:366
+#, c-format
+msgid ""
+"Usage: su [OPTS] [-] [username [ARGS]]\n"
+"\t-\tmake this a login shell\n"
+"\t-c, --command=<command>\n"
+"\t\tpass command to the invoked shell using its -c\n"
+"\t\toption\n"
+"       -m, -p, --preserve-environment\n"
+"\t\tdo not reset environment variables, and keep the\n"
+"\t\tsame shell\n"
+"       -s, --shell=<shell>\n"
+"\t\tuse shell instead of the default in /etc/passwd\n"
+msgstr ""
+
+#: src/su.c:453
 #, c-format
 msgid "%s: pam_start: error %d\n"
 msgstr ""
 
-#: src/su.c:336
+#: src/su.c:483
 #, c-format
 msgid "Unknown id: %s\n"
 msgstr ""
 
-#: src/su.c:373 src/su.c:390
+#: src/su.c:520 src/su.c:537
 #, c-format
 msgid "You are not authorized to su %s\n"
 msgstr ""
 
 #. require own password
-#: src/su.c:385
+#: src/su.c:532
 msgid "(Enter your own password.)"
 msgstr ""
 
-#: src/su.c:419
+#: src/su.c:574
 #, c-format
 msgid ""
 "%s: %s\n"
 "(Ignored)\n"
 msgstr ""
 
-#: src/su.c:626
-msgid "No shell\n"
-msgstr ""
-
-#: src/suauth.c:100
-msgid "Access to su to that account DENIED.\n"
-msgstr ""
-
-#: src/suauth.c:107
-msgid "Password authentication bypassed.\n"
-msgstr ""
-
-#: src/suauth.c:114
-msgid "Please enter your OWN password as authentication.\n"
-msgstr ""
-
 #. must be a password file!
 #: src/sulogin.c:132
+#, c-format
 msgid "No password file\n"
 msgstr ""
 
@@ -1822,6 +1892,7 @@
 #. * Fail secure
 #.
 #: src/sulogin.c:174
+#, c-format
 msgid "No password entry for 'root'\n"
 msgstr ""
 
@@ -1837,507 +1908,520 @@
 msgid "Entering System Maintenance Mode\n"
 msgstr ""
 
-#: src/useradd.c:251
+#: src/useradd.c:253
 #, c-format
 msgid "%s: rebuild the group database\n"
 msgstr ""
 
-#: src/useradd.c:260
+#: src/useradd.c:262
 #, c-format
 msgid "%s: rebuild the shadow group database\n"
 msgstr ""
 
-#: src/useradd.c:295 src/useradd.c:309 src/usermod.c:981 src/usermod.c:995
+#: src/useradd.c:297 src/useradd.c:311 src/usermod.c:983 src/usermod.c:997
 #, c-format
 msgid "%s: invalid numeric argument `%s'\n"
 msgstr ""
 
-#: src/useradd.c:365
+#: src/useradd.c:367
 #, c-format
 msgid "%s: unknown gid %s\n"
 msgstr ""
 
-#: src/useradd.c:373 src/useradd.c:676 src/useradd.c:1269 src/usermod.c:260
-#: src/usermod.c:1128
+#: src/useradd.c:375 src/useradd.c:678 src/useradd.c:1271 src/usermod.c:262
+#: src/usermod.c:1130
 #, c-format
 msgid "%s: unknown group %s\n"
 msgstr ""
 
-#: src/useradd.c:440
+#: src/useradd.c:442
 #, c-format
 msgid "group=%s,%u  basedir=%s  skel=%s\n"
 msgstr ""
 
-#: src/useradd.c:444
+#: src/useradd.c:446
 #, c-format
 msgid "shell=%s  "
 msgstr ""
 
-#: src/useradd.c:446
+#: src/useradd.c:448
 #, c-format
 msgid "inactive=%ld  expire=%s"
 msgstr ""
 
-#: src/useradd.c:450
+#: src/useradd.c:452
 #, c-format
 msgid "GROUP=%u\n"
 msgstr ""
 
-#: src/useradd.c:451
+#: src/useradd.c:453
 #, c-format
 msgid "HOME=%s\n"
 msgstr ""
 
-#: src/useradd.c:453
+#: src/useradd.c:455
 #, c-format
 msgid "INACTIVE=%ld\n"
 msgstr ""
 
-#: src/useradd.c:454
+#: src/useradd.c:456
 #, c-format
 msgid "EXPIRE=%s\n"
 msgstr ""
 
-#: src/useradd.c:456
+#: src/useradd.c:458
 #, c-format
 msgid "SHELL=%s\n"
 msgstr ""
 
-#: src/useradd.c:457
+#: src/useradd.c:459
 #, c-format
 msgid "SKEL=%s\n"
 msgstr ""
 
-#: src/useradd.c:498
+#: src/useradd.c:500
 #, c-format
 msgid "%s: cannot create new defaults file\n"
 msgstr ""
 
-#: src/useradd.c:503
+#: src/useradd.c:505
 #, c-format
 msgid "%s: cannot open new defaults file\n"
 msgstr ""
 
-#: src/useradd.c:596 src/useradd.c:608
+#: src/useradd.c:598 src/useradd.c:610
 #, c-format
 msgid "%s: rename: %s"
 msgstr ""
 
-#: src/useradd.c:697 src/usermod.c:281
+#: src/useradd.c:699 src/usermod.c:283
 #, c-format
 msgid "%s: group `%s' is a NIS group.\n"
 msgstr ""
 
-#: src/useradd.c:706 src/usermod.c:290
+#: src/useradd.c:708 src/usermod.c:292
 #, c-format
 msgid "%s: too many groups specified (max %d).\n"
 msgstr ""
 
-#: src/useradd.c:738 src/usermod.c:322
+#: src/useradd.c:740 src/usermod.c:324
 #, c-format
 msgid "usage: %s\t[-u uid [-o]] [-g group] [-G group,...] \n"
 msgstr ""
 
-#: src/useradd.c:742
+#: src/useradd.c:744
+#, c-format
 msgid "\t\t[-d home] [-s shell] [-c comment] [-m [-k template]]\n"
 msgstr ""
 
-#: src/useradd.c:745 src/usermod.c:329
+#: src/useradd.c:747 src/usermod.c:331
+#, c-format
 msgid "[-f inactive] [-e expire ] "
 msgstr ""
 
-#: src/useradd.c:748
+#: src/useradd.c:750
+#, c-format
 msgid "[-A program] "
 msgstr ""
 
-#: src/useradd.c:750
+#: src/useradd.c:752
+#, c-format
 msgid "[-p passwd] name\n"
 msgstr ""
 
-#: src/useradd.c:753
+#: src/useradd.c:755
 #, c-format
 msgid "       %s\t-D [-g group] [-b base] [-s shell]\n"
 msgstr ""
 
-#: src/useradd.c:756
+#: src/useradd.c:758
+#, c-format
 msgid "\t\t[-f inactive] [-e expire ]\n"
 msgstr ""
 
-#: src/useradd.c:850 src/usermod.c:480
+#: src/useradd.c:852 src/usermod.c:482
 #, c-format
 msgid "%s: error locking group file\n"
 msgstr ""
 
-#: src/useradd.c:855 src/usermod.c:486
+#: src/useradd.c:857 src/usermod.c:488
 #, c-format
 msgid "%s: error opening group file\n"
 msgstr ""
 
-#: src/useradd.c:862 src/usermod.c:597
+#: src/useradd.c:864 src/usermod.c:599
 #, c-format
 msgid "%s: error locking shadow group file\n"
 msgstr ""
 
-#: src/useradd.c:867 src/usermod.c:603
+#: src/useradd.c:869 src/usermod.c:605
 #, c-format
 msgid "%s: error opening shadow group file\n"
 msgstr ""
 
-#: src/useradd.c:1039
+#: src/useradd.c:1041
 #, c-format
 msgid "%s: uid %u is not unique\n"
 msgstr ""
 
-#: src/useradd.c:1069
+#: src/useradd.c:1071
 #, c-format
 msgid "%s: can't get unique uid\n"
 msgstr ""
 
-#: src/useradd.c:1174 src/useradd.c:1325 src/usermod.c:1075 src/usermod.c:1086
-#: src/usermod.c:1096 src/usermod.c:1143 src/usermod.c:1187
+#: src/useradd.c:1176 src/useradd.c:1327 src/usermod.c:1077 src/usermod.c:1088
+#: src/usermod.c:1098 src/usermod.c:1145 src/usermod.c:1189
 #, c-format
 msgid "%s: invalid field `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1189
+#: src/useradd.c:1191
 #, c-format
 msgid "%s: invalid base directory `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1199
+#: src/useradd.c:1201
 #, c-format
 msgid "%s: invalid comment `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1210
+#: src/useradd.c:1212
 #, c-format
 msgid "%s: invalid home directory `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1229 src/usermod.c:1110
+#: src/useradd.c:1231 src/usermod.c:1112
 #, c-format
 msgid "%s: invalid date `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1242
+#: src/useradd.c:1244
 #, c-format
 msgid "%s: shadow passwords required for -e\n"
 msgstr ""
 
-#: src/useradd.c:1258
+#: src/useradd.c:1260
 #, c-format
 msgid "%s: shadow passwords required for -f\n"
 msgstr ""
 
-#: src/useradd.c:1336
+#: src/useradd.c:1338
 #, c-format
 msgid "%s: invalid shell `%s'\n"
 msgstr ""
 
-#: src/useradd.c:1377
+#: src/useradd.c:1379
 #, c-format
 msgid "%s: invalid user name '%s'\n"
 msgstr ""
 
-#: src/useradd.c:1413 src/userdel.c:298 src/usermod.c:1255
+#: src/useradd.c:1415 src/userdel.c:309 src/usermod.c:1257
 #, c-format
 msgid "%s: cannot rewrite password file\n"
 msgstr ""
 
-#: src/useradd.c:1420 src/userdel.c:303 src/usermod.c:1262
+#: src/useradd.c:1422 src/userdel.c:314 src/usermod.c:1264
 #, c-format
 msgid "%s: cannot rewrite shadow password file\n"
 msgstr ""
 
-#: src/useradd.c:1461 src/userdel.c:367 src/usermod.c:1296
+#: src/useradd.c:1463 src/userdel.c:378 src/usermod.c:1298
 #, c-format
 msgid "%s: unable to lock password file\n"
 msgstr ""
 
-#: src/useradd.c:1466 src/userdel.c:372 src/usermod.c:1301
+#: src/useradd.c:1468 src/userdel.c:383 src/usermod.c:1303
 #, c-format
 msgid "%s: unable to open password file\n"
 msgstr ""
 
-#: src/useradd.c:1474 src/userdel.c:379 src/usermod.c:1308
+#: src/useradd.c:1476 src/userdel.c:390 src/usermod.c:1310
 #, c-format
 msgid "%s: cannot lock shadow password file\n"
 msgstr ""
 
-#: src/useradd.c:1481 src/userdel.c:385 src/usermod.c:1314
+#: src/useradd.c:1483 src/userdel.c:396 src/usermod.c:1316
 #, c-format
 msgid "%s: cannot open shadow password file\n"
 msgstr ""
 
-#: src/useradd.c:1580 src/usermod.c:1405
+#: src/useradd.c:1582 src/usermod.c:1407
 #, c-format
 msgid "%s: error adding authentication method\n"
 msgstr ""
 
-#: src/useradd.c:1604
+#: src/useradd.c:1606
 #, c-format
 msgid "%s: error adding new password entry\n"
 msgstr ""
 
-#: src/useradd.c:1618
+#: src/useradd.c:1620
 #, c-format
 msgid "%s: error updating password dbm entry\n"
 msgstr ""
 
-#: src/useradd.c:1634 src/usermod.c:1471
+#: src/useradd.c:1636 src/usermod.c:1473
 #, c-format
 msgid "%s: error adding new shadow password entry\n"
 msgstr ""
 
-#: src/useradd.c:1650 src/usermod.c:1488
+#: src/useradd.c:1652 src/usermod.c:1490
 #, c-format
 msgid "%s: error updating shadow passwd dbm entry\n"
 msgstr ""
 
-#: src/useradd.c:1682
+#: src/useradd.c:1684
 #, c-format
 msgid "%s: cannot create directory %s\n"
 msgstr ""
 
-#: src/useradd.c:1806 src/usermod.c:1233
+#: src/useradd.c:1808 src/usermod.c:1235
 #, c-format
 msgid "%s: user %s exists\n"
 msgstr ""
 
-#: src/useradd.c:1821
+#: src/useradd.c:1823
 #, c-format
 msgid ""
 "%s: group %s exists - if you want to add this user to that group, use -g.\n"
 msgstr ""
 
-#: src/useradd.c:1854
+#: src/useradd.c:1856
 #, c-format
 msgid "%s: warning: CREATE_HOME not supported, please use -m instead.\n"
 msgstr ""
 
-#: src/userdel.c:129
+#: src/userdel.c:131
 #, c-format
 msgid "usage: %s [-r] name\n"
 msgstr ""
 
-#: src/userdel.c:181 src/userdel.c:267
+#: src/userdel.c:184 src/userdel.c:274
 #, c-format
 msgid "%s: error updating group entry\n"
 msgstr ""
 
-#: src/userdel.c:191 src/userdel.c:276
+#: src/userdel.c:194 src/userdel.c:283
 #, c-format
 msgid "%s: cannot update dbm group entry\n"
 msgstr ""
 
-#: src/userdel.c:219
+#: src/userdel.c:226
 #, c-format
 msgid "%s: cannot remove dbm group entry\n"
 msgstr ""
 
-#: src/userdel.c:308
+#: src/userdel.c:319
 #, c-format
 msgid "%s: cannot rewrite TCFS key file\n"
 msgstr ""
 
-#: src/userdel.c:392
+#: src/userdel.c:403
 #, c-format
 msgid "%s: cannot lock TCFS key file\n"
 msgstr ""
 
-#: src/userdel.c:397
+#: src/userdel.c:408
 #, c-format
 msgid "%s: cannot open TCFS key file\n"
 msgstr ""
 
-#: src/userdel.c:408
+#: src/userdel.c:419
 #, c-format
 msgid "%s: cannot open group file\n"
 msgstr ""
 
-#: src/userdel.c:419
+#: src/userdel.c:430
 #, c-format
 msgid "%s: cannot open shadow group file\n"
 msgstr ""
 
-#: src/userdel.c:451 src/userdel.c:468
+#: src/userdel.c:462 src/userdel.c:479
 #, c-format
 msgid "%s: error deleting authentication\n"
 msgstr ""
 
-#: src/userdel.c:479
+#: src/userdel.c:490
 #, c-format
 msgid "%s: error deleting password entry\n"
 msgstr ""
 
-#: src/userdel.c:485
+#: src/userdel.c:496
 #, c-format
 msgid "%s: error deleting shadow password entry\n"
 msgstr ""
 
-#: src/userdel.c:496
+#: src/userdel.c:507
 #, c-format
 msgid "%s: error deleting TCFS entry\n"
 msgstr ""
 
-#: src/userdel.c:510
+#: src/userdel.c:521
 #, c-format
 msgid "%s: error deleting password dbm entry\n"
 msgstr ""
 
-#: src/userdel.c:533
+#: src/userdel.c:544
 #, c-format
 msgid "%s: error deleting shadow passwd dbm entry\n"
 msgstr ""
 
-#: src/userdel.c:569
+#: src/userdel.c:580
 #, c-format
 msgid "%s: user %s is currently logged in\n"
 msgstr ""
 
-#: src/userdel.c:659
+#: src/userdel.c:670
 #, c-format
 msgid "%s: warning: %s not owned by %s, not removing\n"
 msgstr ""
 
-#: src/userdel.c:666
+#: src/userdel.c:677
 #, c-format
 msgid "%s: warning: can't remove "
 msgstr ""
 
-#: src/userdel.c:758 src/usermod.c:1023
+#: src/userdel.c:769 src/usermod.c:1025
 #, c-format
 msgid "%s: user %s does not exist\n"
 msgstr ""
 
-#: src/userdel.c:770 src/usermod.c:1039
+#: src/userdel.c:781 src/usermod.c:1041
 #, c-format
 msgid "%s: user %s is a NIS user\n"
 msgstr ""
 
-#: src/userdel.c:799
+#: src/userdel.c:810
 #, c-format
 msgid "%s: %s not owned by %s, not removing\n"
 msgstr ""
 
-#: src/userdel.c:818
+#: src/userdel.c:829
 #, c-format
 msgid "%s: not removing directory %s (would remove home of user %s)\n"
 msgstr ""
 
-#: src/userdel.c:830
+#: src/userdel.c:841
 #, c-format
 msgid "%s: error removing directory %s\n"
 msgstr ""
 
-#: src/usermod.c:326
+#: src/usermod.c:328
+#, c-format
 msgid "\t\t[-d home [-m]] [-s shell] [-c comment] [-l new_name]\n"
 msgstr ""
 
-#: src/usermod.c:332
+#: src/usermod.c:334
+#, c-format
 msgid "[-A {DEFAULT|program},... ] "
 msgstr ""
 
-#: src/usermod.c:334
+#: src/usermod.c:336
+#, c-format
 msgid "[-p passwd] [-L|-U] name\n"
 msgstr ""
 
-#: src/usermod.c:514
+#: src/usermod.c:516
 #, c-format
 msgid "%s: out of memory in update_group\n"
 msgstr ""
 
-#: src/usermod.c:641
+#: src/usermod.c:643
 #, c-format
 msgid "%s: out of memory in update_gshadow\n"
 msgstr ""
 
-#: src/usermod.c:1210
+#: src/usermod.c:1212
 #, c-format
 msgid "%s: no flags given\n"
 msgstr ""
 
-#: src/usermod.c:1217
+#: src/usermod.c:1219
 #, c-format
 msgid "%s: shadow passwords required for -e and -f\n"
 msgstr ""
 
-#: src/usermod.c:1239
+#: src/usermod.c:1241
 #, c-format
 msgid "%s: uid %lu is not unique\n"
 msgstr ""
 
-#: src/usermod.c:1392
+#: src/usermod.c:1394
 #, c-format
 msgid "%s: error deleting authentication method\n"
 msgstr ""
 
-#: src/usermod.c:1418
+#: src/usermod.c:1420
 #, c-format
 msgid "%s: error changing authentication method\n"
 msgstr ""
 
-#: src/usermod.c:1436
+#: src/usermod.c:1438
 #, c-format
 msgid "%s: error changing password entry\n"
 msgstr ""
 
-#: src/usermod.c:1442
+#: src/usermod.c:1444
 #, c-format
 msgid "%s: error removing password entry\n"
 msgstr ""
 
-#: src/usermod.c:1451
+#: src/usermod.c:1453
 #, c-format
 msgid "%s: error adding password dbm entry\n"
 msgstr ""
 
-#: src/usermod.c:1459
+#: src/usermod.c:1461
 #, c-format
 msgid "%s: error removing passwd dbm entry\n"
 msgstr ""
 
-#: src/usermod.c:1478
+#: src/usermod.c:1480
 #, c-format
 msgid "%s: error removing shadow password entry\n"
 msgstr ""
 
-#: src/usermod.c:1495
+#: src/usermod.c:1497
 #, c-format
 msgid "%s: error removing shadow passwd dbm entry\n"
 msgstr ""
 
-#: src/usermod.c:1524
+#: src/usermod.c:1526
 #, c-format
 msgid "%s: directory %s exists\n"
 msgstr ""
 
-#: src/usermod.c:1533
+#: src/usermod.c:1535
 #, c-format
 msgid "%s: can't create %s\n"
 msgstr ""
 
-#: src/usermod.c:1539
+#: src/usermod.c:1541
 #, c-format
 msgid "%s: can't chown %s\n"
 msgstr ""
 
-#: src/usermod.c:1556
+#: src/usermod.c:1553
+#, c-format
+msgid "%s: warning: failed to completely remove old home directory %s"
+msgstr ""
+
+#: src/usermod.c:1563
 #, c-format
 msgid "%s: cannot rename directory %s to %s\n"
 msgstr ""
 
 #. better leave it alone
-#: src/usermod.c:1653
+#: src/usermod.c:1660
 #, c-format
 msgid "%s: warning: %s not owned by %s\n"
 msgstr ""
 
-#: src/usermod.c:1659
+#: src/usermod.c:1666
 msgid "failed to change mailbox owner"
 msgstr ""
 
-#: src/usermod.c:1667
+#: src/usermod.c:1674
 msgid "failed to rename mailbox"
 msgstr ""
 
@@ -2362,6 +2446,7 @@
 msgstr ""
 
 #: src/vipw.c:244
+#, c-format
 msgid ""
 "Usage:\n"
 "`vipw' edits /etc/passwd        `vipw -s' edits /etc/shadow\n"
